spring-cloud-app-broker-deployer-cloudfoundry/src/main/java/org/springframework/cloud/appbroker/deployer/cloudfoundry/CloudFoundryDeploymentProperties.java

```diff
@@ -97,10 +99,15 @@
 	private boolean enableRandomAppNamePrefix = true;
 
 	/**
-	 * Timeout for blocking API calls, in seconds.
+	 * Timeout for blocking CF API calls, in seconds.
 	 */
 	private long apiTimeout = 360L;
 
+	/**
+	 * Timeout for polled async CF API calls, in seconds. Named "completionTimeout" in cf-java-client
+	 */
+	private long apiPollingTimeout = DEFAULT_API_POLLING_TIMEOUT_SECONDS;
```

Comment:
Since we have `api-timeout` which relates to the backing application timeout and it is documented as
```
The timeout value used for blocking CF API calls, in seconds.
```

Adding now `api-polling-timeout` feels like it is related to backing applications when it isn't. Shall we set a name that is more explicit to backing services to make the difference more obvious?
Something in the line of: `backingServiceCompletionTimeout`. What do you think?

Reply:
Many other CF java client requests have a configurable completion timeout beyond services, even though they are not net leveraged in SCAB e.g.:
- [DeleteApplicationRequest](https://github.com/cloudfoundry/cf-java-client/blob/bb3de77403d7c79daab360d715cc6113e90c2ce7/cloudfoundry-operations/src/main/java/org/cloudfoundry/operations/applications/_DeleteApplicationRequest.java#L34)
- [DeleteSpaceRequest](https://github.com/cloudfoundry/cf-java-client/blob/bb3de77403d7c79daab360d715cc6113e90c2ce7/cloudfoundry-operations/src/main/java/org/cloudfoundry/operations/spaces/_DeleteSpaceRequest.java#L33)
- [DeleteRouteRequest](https://github.com/cloudfoundry/cf-java-client/blob/bb3de77403d7c79daab360d715cc6113e90c2ce7/cloudfoundry-operations/src/main/java/org/cloudfoundry/operations/routes/_DeleteRouteRequest.java#L41)

By naming the property generically (and not specifically to services), I was hoping this property could also be applied to other cf-java-client requests in the future

Reply:
I see it now. Thanks for explaining!
If we wanted to introduce some of the more specific, we could do something like:
by default use the value of `DEFAULT_API_POLLING_TIMEOUT_SECONDS`
if a value was provided for `api-polling-timeout` use that.
if a value for a specific request was provided, overwrite `api-polling-timeout` with that new property value.

Reply:
Thanks! Your proposal to use api polling timeout as a default for future more specific properties makes sense. I'm therefore resolving the conversation, please reopen if there is something missing.